web/jobs/jobs.go

```diff
@@ -259,7 +277,30 @@ func contactSupport(c echo.Context) error {
 	return c.NoContent(http.StatusNoContent)
 }
 
-func newTrigger(c echo.Context) error {
+func (h *HTTPHandler) sendCampaignEmail(c echo.Context) error {
+	inst := middlewares.GetInstance(c)
+
+	if err := middlewares.Allow(c, permission.POST, &job.JobRequest{WorkerType: "sendmail"}); err != nil {
+		return err
+	}
+
+	req := apiCampaign{}
+	if _, err := jsonapi.Bind(c.Request().Body, &req); err != nil {
+		return wrapJobsError(err)
+	}
+
+	err := h.emailer.SendCampaignEmail(inst, &emailer.SendEmailCmd{
+		Subject: req.Arguments.Subject,
+		Parts:   req.Arguments.Parts,
+	})
+	if err != nil {
+		return err
```

Comment:
I think the response will be a 500 when the subject or the parts are missing, instead of a 4xx. We should fix that by wrapping the error.

Reply:
Won't we get an error when unmarshalling the request since the fields don't have the `omitempty` option?